src/d3d9/d3d9_device.cpp

```diff
@@ -1621,6 +1625,9 @@ namespace dxvk {
           1u
         };
 
+        if (rectExtent.width == 0 || rectExtent.height == 0) {
```

Comment:
this should probably check `pRects[i].x2 <= rectOffset.x` etc to be more robust.